cc-pk-service/src/handleData.ts

```diff
@@ -1,121 +1,74 @@
-
+/* eslint-disable  @typescript-eslint/no-explicit-any */
 import axios from 'axios';
-import dotenv from 'dotenv';
-import { Survey, ItemInfo, Item, Building } from './types';
-
-dotenv.config();
-
-// Variables
-const username: string = process.env.PK_USERNAME ?? '';
-const password: string = process.env.PK_PASSWORD ?? '';
-const c_id: string = process.env.PK_CLIENTID ?? '';
-const g_type: string = process.env.PK_GRANTTYPE ?? '';
-
-let userToken:string;
-export let userId:string;
-
-// Get usertoken and userid
-getUserInfo();
+import { Survey, PkItem, Item, Building, ItemInfo, ReusableMaterial } from './types';
 
 /**
  * Initalize an empty ItemInfo -object
  * @returns An empty ItemInfo -object
  */
-function initInfo() {
+function initItemInfo() {
     const info:ItemInfo = {
-        id:'',
-        componentName:'',
-        reusableMaterialId:'',
-        usability:'',
-        amount:0,
-        unit:'',
-        description:'',
-        images: '',
-        amountAsWaste:0,
-        metadata:{
-            creatorId:'',
-            createdAt:'',
-            lastModifierId:'',
-            modifiedAt: ''
-        }
+        title: '',
+        material: '',
+        unit: '',
+        amount: 0,
+        streetAddress: '',
+        zipCode: '',
+        area: '',
+        images: ''
     }
     return info;
 }
-
-/**
- * Collect user info at the beginning
- */
-async function getUserInfo() {
-    userToken = await getToken();
-    userId = await getId();
-    //console.log(userId); // Use this to check your usedId
-}
-
-/**
-* Get token for the current user, so accessing the Purkukartoitus API is possible
-* @param user Username
-* @param pw Password
-* @returns Token for accessing the Purkukartoitus API
-*/
-export async function getToken() {
-    try {
-        const response = axios.post(
-            'https://auth.purkukartoitus.fi/auth/realms/rapurc/protocol/openid-connect/token',
-            new URLSearchParams({
-                'client_id': c_id,
-                'grant_type': g_type,
-                'username': username,
-                'password': password
-        }));
-        return (await response).data.access_token;
-    } catch (err) {
-        console.log('Error: cannot fetch id for the current user');
-        return err;
-    }
-}
```

Comment:
If doing a new getToken -function for splitting and returning, you could add it here. Code snippet also from this in teams chat